rotkehlchen/exchanges/kraken.py

```diff
@@ -225,6 +239,19 @@ def history_event_from_kraken(
                 f'Failed to read ledger event from kraken {raw_event} due to {msg}',
             )
             return [], False
+
+    for event_set in receive_spend_events.values():
+        if len(event_set) == 2:
+            for event in event_set:
```

Comment:
If it is guaranteed here that `spend` event goes first and `receive` event goes after, then I would access `event_set` by index instead of looping, since the order is known beforehand.

If there are no guarantees on the order, then I guess there is currently a bug, and we have to make sure that the spend event goes first.

I would solve this by doing
```python
spend_event, receive_event = event_set
if spend_event.event_type != HistoryEventSubType.SPEND:
    spend_event, receive_event = receive_event, spend_event

spend_event.balance.amount = -spend_event.balance.amount
spend_event.event_type = HistoryEventType.TRADE
receive_event.event_type = HistoryEventType.TRADE
spend_event.event_subtype = HistoryEventSubType.SPEND
receive_event.event_subtype = HistoryEventSubType.RECEIVE
```

Does this make sense to you?

Reply:
Commmented already in discord. We use another approach to build the trades since the logic is more complicated